### ISSUE TITLE
Update CC -> SF field mapping

### ISSUE THREAD

**Issue by ritazagoni** [OpenFn/Miracle-Feet#105]
## Background, context, and business value

We are updating the `Visit.Brace_Problems_Type__c` field in Salesforce, but we don't have a mapping table in place to update the SF values.

## The specific request, in as few words as possible

Jobs to add new mapping to:
1. [Add New Visit](https://github.com/OpenFn/Miracle-Feet/blob/master/sample_data/stateUpdateVisit.json)
2. [Update Visit
](https://github.com/OpenFn/Miracle-Feet/blob/master/update_visit__UPDATED.js)
Mapping to be added:

```
const braceProblemsTypeMap = {
feet_slipping: 'Feet Slipping',
none: 'None',
motion_range: 'Motion Range',
not_wearing_enough: 'Not Wearing Enough'
child_not_tolerating: 'Child Not Tolerating',
family_not_accepting: 'Family Not Accepting',
incorrect_shoe_size: 'Incorrect Shoe Size',
skin_irritation: 'Skin Irritation',
broken_defective_brace: 'Broken Defective Brace',
other: 'Other'
  };
```
Steps:
1. Update [existing `Visit.Brace_Problems_Type__c`](https://github.com/OpenFn/Miracle-Feet/blob/ac9d36e504940a208296f7115feb932d4369d1bc/add_new_visit__UPDATED.js#L244-L245) mapping in Add new visit to use mapping table
2. Add same `Brace_Problems_Type__c`` mapping to `Contact` object in [Add new visit](https://github.com/OpenFn/Miracle-Feet/blob/master/add_new_visit__UPDATED.js)
3. Update [existing `Visit.Brace_Problems_Type__c`](https://github.com/OpenFn/Miracle-Feet/blob/ac9d36e504940a208296f7115feb932d4369d1bc/update_visit__UPDATED.js#L203-L204) mapping in Update Visit to use mapping tabel
4. Add same `Brace_Problems_Type__c` mapping to `Contact` in [Update Visit](https://github.com/OpenFn/Miracle-Feet/blob/master/update_visit__UPDATED.js)

## state.json

See LP `MiracleFeet Salesforce Production`
Example state for [1. Update Visit](https://github.com/OpenFn/Miracle-Feet/blob/master/sample_data/stateUpdateVisit.json)
Example state for [2. Add New Visit](https://github.com/OpenFn/Miracle-Feet/blob/master/sample_data/log_new_visit_multiple_brace_problems_type.json)

## adaptor

`language-salesforce`

**Comment by aleksa-krolls** [comment]
@elias-ba Next issue to pick up this morning whenever you have a minute... liaise with @ritazagoni for any questions or additional info needed, thanks :D

**Comment by elias-ba** [comment]
Taking this one now.

**Comment by aleksa-krolls** [comment]
@elias-ba let @ritazagoni know if any questions pop up... she'll be online for a couple more hours.

**Comment by elias-ba** [comment]
Yes @aleksa-krolls, I am currently following the steps outlined there. Will reach out for any question.

**Comment by elias-ba** [comment]
@ritazagoni I think the link for example state 1 is not the right one, could you confirm ?

**Comment by aleksa-krolls** [comment]
@elias-ba I think this is the right link for state 1: https://github.com/OpenFn/Miracle-Feet/blob/master/sample_data/log_new_visit_multiple_brace_problems_type.json

@ritazagoni to confirm tomorrow

**Comment by elias-ba** [comment]
Thanks @aleksa-krolls. Let me try to test with that one.

**Comment by elias-ba** [comment]
@aleksa-krolls I think we'll need to wait for @ritazagoni to give us the right state to use for state 1. I have tested the one you gave me above but it produce this error `TypeError [Error]: Cannot destructure property 'clinic_code' of 'state.data.properties' as it is undefined.`. I suspect that it's because it's not the right state.

I am now moving to the second item in the checklist.

**Comment by ritazagoni** [comment]
@elias-ba the sample data is updated!

**Comment by ritazagoni** [comment]
hey @elias-ba yes `brace_problems_type` can be either one value or multiple values separated by ` `.
Here's an example of how we handle this field in another job: https://github.com/OpenFn/Miracle-Feet/blob/9aa7e5f585ecb33c338348b5dfd6907bd29b2935/schedule_sms.js#L302-L305

**Comment by elias-ba** [comment]
Thanks @ritazagoni. How do we materialize that using the Mapping above. In the case of `brace_problems_type` having more than one value, do we concatenate the different values of the mapping or do we get an array of the values ?

**Comment by ritazagoni** [comment]
@elias-ba We have to use the [handleMultiSelect](https://github.com/OpenFn/Miracle-Feet/blob/ac9d36e504940a208296f7115feb932d4369d1bc/add_new_visit__UPDATED.js#L244-L245) function to update multiple values. maybe @lakhassane can help with how to use it?

**Comment by elias-ba** [comment]
Thanks @ritazagoni for mentioning that. Lemme see if I can pair with @lakhassane.

**Comment by elias-ba** [comment]
@ritazagoni this is now working fine. The mapping is integrated in the two expressions and all the tests are successful. Please do your tests and let me know if you have any feed-backs.

Sorry for the time I spent on this, these job expression are not easy to read and I was making sure everything got integrated very well.

Thanks ++

**Comment by ritazagoni** [comment]
no worries and thanks @elias-ba !

**Comment by ritazagoni** [comment]
@elias-ba could you check if these erros are related? https://openfn.org/projects/miraclefeet/runs/061b9feb-6cd2-7a05-8d99-4720c5664054 and https://openfn.org/projects/miraclefeet/runs/061b9feb-6ce4-7e71-a69c-e06485be151e ?

**Comment by elias-ba** [comment]
@ritazagoni I just pushed a correction. Please check if this solved !

**Comment by ritazagoni** [comment]
Looking great thanks @elias-ba !
